imageio/core/util.py

```diff
@@ -46,39 +47,69 @@ def urlopen(*args, **kwargs):
     return urlopen(*args, **kwargs)
 
 
-def image_as_uint8(im):
-    """ Convert the given image to uint8
-    
-    If the dtype is already uint8, it is returned as-is. If the image
-    is float, and all values are between 0 and 1, the values are
-    multiplied by 255. In all other situations, the values are scaled
-    such that the minimum value becomes 0 and the maximum value becomes
-    255.
+def image_as_uint(img_in, bitdepth=None):
+    """ Convert the given image to uint (default: uint8)
+    
+    If the dtype already matches the desired format, it is returned
+    as-is. If the image is float, and all values are between 0 and 1,
+    the values are multiplied by np.power(2.0, bitdepth). In all other
+    situations, the values are scaled such that the minimum value
+    becomes 0 and the maximum value becomes np.power(2.0, bitdepth)-1
+    (255 for 8-bit and 65535 for 16-bit).
     """
-    if not isinstance(im, np.ndarray):
-        raise ValueError('image must be a numpy array')
+    if not bitdepth:
+        bitdepth = 8
+    if not isinstance(img_in, np.ndarray):
+        raise ValueError('Image must be a numpy array')
+    if bitdepth == 8:
+        out_type = np.uint8
+    elif bitdepth == 16:
+        out_type = np.uint16
+    else:
+        raise ValueError('Bitdepth must be either 8 or 16')
+    im = img_in.copy()
```

Comment:
Great catch, the im *= ... is in-place while im = im * ... works on a copy. Fixed.